ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/StreamedQueryResult.java

```diff
@@ -90,4 +93,9 @@
    */
   boolean isFailed();
 
+  boolean hasContinuationToken();
+
+  Optional<AtomicReference<String>> getContinuationToken();
+
+  CompletableFuture<StreamedQueryResult> retry();
```

Comment:
We should definitely add comments for this method to make clear when it should be used.  If there's no continuation token or the feature is disabled, normal push queries are used or it's a pull query, it will just issue the request as normal, so it shouldn't require more introspection in practice.  Not sure if it makes sense to call it `continue()` or maybe `retryOrContinue()`.

Reply:
added javadoc